go/cmd/server/server.go

```diff
@@ -49,21 +52,87 @@ func (s *servingServiceServer) GetOnlineFeatures(ctx context.Context, request *s
 			FeatureNames: &serving.FeatureList{Val: make([]string, 0)},
 		},
 	}
+	// Entities are currently part of the features as a value and the order that we add it to the resp MetaData
+	// Need to figure out a way to map the correct entities to the correct ordering
+	entityValuesMap := make(map[string][]*prototypes.Value, 0)
+	featureNames := make([]string, len(featureVectors))
+	for idx, vector := range featureVectors {
 
-	for _, vector := range featureVectors {
 		resp.Metadata.FeatureNames.Val = append(resp.Metadata.FeatureNames.Val, vector.Name)
-
+		featureNames[idx] = vector.Name
 		values, err := types.ArrowValuesToProtoValues(vector.Values)
 		if err != nil {
 			return nil, err
 		}
-
+		if _, ok := request.Entities[vector.Name]; ok {
+			entityValuesMap[vector.Name] = values
+		}
 		resp.Results = append(resp.Results, &serving.GetOnlineFeaturesResponse_FeatureVector{
 			Values:          values,
 			Statuses:        vector.Statuses,
 			EventTimestamps: vector.Timestamps,
 		})
 	}
 
+	go generateLogs(s, entityValuesMap, featureNames, resp.Results, request.RequestContext)
 	return resp, nil
 }
+
+func generateLogs(s *servingServiceServer, entityMap map[string][]*prototypes.Value, featureNames []string, features []*serving.GetOnlineFeaturesResponse_FeatureVector, requestData map[string]*prototypes.RepeatedValue) error {
+	// Add a log with the request context
+	if len(requestData) > 0 {
+		requestContextLog := logging.Log{
+			RequestContext: requestData,
+		}
+		s.loggingService.EmitLog(&requestContextLog)
+	}
+
+	if len(features) <= 0 {
+		return nil
+	}
+
+	entityArr, err := s.fs.ListEntities(true)
+	if err != nil {
+		return err
+	}
+
+	joinKeys := make([]string, 0)
+	for _, entity := range entityArr {
```

Comment:
we shouldn't use all entities here, but only entities that required by specific feature service

Reply:
Ideally in this function we should already have schema generated from feature service and use it as source of truth for columns/fields names and order

Reply:
Addressing 1, refer to other comment about entities aren't tied to a specific feature service right now in our data model. 
Addressing 2, I think I implemented it this way because of the way the request is structured. A featureservice request contains a projection that contains the features in order as a slice. This is processed and returned in the a similar way via slices so as long as I refer to featureservice as the source of truth for the ordering of the field names, i feel like it makes more sense this way. Wdyt?